pkg/minikube/bootstrapper/certs.go

```diff
@@ -144,9 +152,9 @@ func generateCerts(k8s KubernetesConfig) error {
 		{ // apiserver serving cert
 			certPath:       filepath.Join(localPath, "apiserver.crt"),
 			keyPath:        filepath.Join(localPath, "apiserver.key"),
-			subject:        "minikube",
-			ips:            []net.IP{net.ParseIP(k8s.NodeIP), serviceIP, net.ParseIP("10.0.0.1")},
-			alternateNames: util.GetAlternateDNS(k8s.DNSDomain),
+			subject:        k8s.APIServerName,
```

Comment:
Can you explain why the "subject" needs to match? Can it stay as "minikube"?

Reply:
The 'subject' change is according to the original issue description. 

But, the goal for this change is to make apiserver accessible outside the machine, and obviously, due to the deprecated usage of Subject CN matching and only check Subject Alternative Names, this is not necessary any more. Will change it back to 'minikube'.